nixos/modules/programs/sway-beta.nix

```diff
@@ -20,13 +29,30 @@ in {
       '';
     };
 
+    extraSessionCommands = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        export SDL_VIDEODRIVER=wayland
+        # needs qt5.qtwayland in systemPackages
+        export QT_QPA_PLATFORM=wayland
+        export QT_WAYLAND_DISABLE_WINDOWDECORATION="1"
+        # Fix for some Java AWT applications (e.g. Android Studio),
+        # use this if they aren't displayed properly:
+        export _JAVA_AWT_WM_NONREPARENTING=1
```

Comment:
Is there any reason to not be doing `GDK_BACKEND=wayland` as well?

Reply:
@colemickens Should be picked by default https://github.com/swaywm/sway/wiki/Running-programs-natively-under-wayland#gtk-30

Reply:
Hm okay. It seems like emersion's most recent edit means that the `QT_WAYLAND_DISABLE_WINDOWDECORATION` isn't needed either. Or have you observed that it is needed?

Reply:
@colemickens Tested on nixos-unstable. Both settings are still needed:
https://github.com/swaywm/sway/wiki#disabling-client-side-qt-decorations
Don't know why its strikethrough there. Maybe that settings is no more required with recent Qt release we don't have.
https://github.com/swaywm/sway/wiki/Running-programs-natively-under-wayland#qt5